src/atmos_param/cloud_simple/cloud_simple.F90

```diff
@@ -0,0 +1,378 @@
+module cloud_simple_mod
+
+#ifdef INTERNAL_FILE_NML
+    use mpp_mod, only: input_nml_file
+#else
+    use fms_mod, only: open_namelist_file, close_file
+#endif
+
+  use            fms_mod, only: stdlog, FATAL, WARNING, error_mesg
+  use   time_manager_mod, only: time_type
+  use sat_vapor_pres_mod, only: compute_qs
+  use      constants_mod, only: KELVIN
+
+  use diag_manager_mod, only: register_diag_field, send_data
+
+  implicit none
+
+  logical ::   do_init = .true. ! update to false after init has been run
+
+  real    ::   cca_lower_limit =  0.0 ! simple convective cloud fraction min
+                                      ! the default is zero. Not being used but 
+                                      ! could be adapted in future.
+
+  ! There were two testing schenarios developed for the sppookie-2 project.
+  ! The first was implemented but not tested very much. The second protocol is
+  ! what was used for that project and what has been more tested in Isca.
+
+  integer ::  spookie_protocol = 2
+
+  ! Critical RH (fraction) values - spookie-2 protocol version 1
+  real    ::   rhc_sfc     = 1.0
+  real    ::   rhc_base    = 0.7
+  real    ::   rhc_top     = 0.3
+
+  ! Critical RH (fraction) values - spookie-2 protocol version 2
+  ! initial values for RH. Updated in calc_rh_min_max
+  real    ::   rh_min_sfc  = 1.0
+  real    ::   rh_min_base = 0.8
+  real    ::   rh_min_top  = 0.9
+
+  real    ::   rh_max_sfc  = 1.0
+  real    ::   rh_max_base = 1.0
+  real    ::   rh_max_top  = 1.0
+
+  ! Pressure (Pa) at cloud bottom and top (very approx)
+  real    ::   p_base     = 70000.
+  real    ::   p_top      = 20000.
+
+  namelist /cloud_simple_nml/  cca_lower_limit, rhc_sfc, rhc_base, rhc_top, &
+                               rh_min_top, rh_min_sfc, rh_min_base,    &
+                               rh_max_top, rh_max_sfc, rh_max_base
+      
+  integer :: id_cf, id_reff_rad, id_frac_liq, id_qcl_rad, id_rh_in_cf, &
+             id_simple_rhcrit, id_rh_min
+ 
+  character(len=14), parameter ::   mod_name_cld = "cloud_simple"
+
+  contains
+
+  !-----------------------------------------------
+
+
+  subroutine cloud_simple_init (axes, Time)
+
+    type(time_type), intent(in)       :: Time
+    integer, intent(in), dimension(4) :: axes
+
+    integer :: io, stdlog_unit
+
+
+#ifdef INTERNAL_FILE_NML
+       read (input_nml_file, nml=cloud_simple_nml, iostat=io)
+#else
+       if ( file_exist('input.nml') ) then
+          nml_unit = open_namelist_file()
```

Comment:
Hi Penny, it looks like you haven't defined `nml_unit` yet, and perhaps define it as an integer variable in line `68`.